### PR TITLE
PP-428: RelayHub tests - Refactor to use ethers and hardhat 

### PR DESCRIPTION
**Pull request by kenny08gt**
## What

- Update the relayhub tests to run with hardhat and ethers js

## Why

- Part of update [rif relay contracts](https://rsklabs.atlassian.net/browse/PP-329#icft=PP-329)

## Refs

- [PP-329](https://rsklabs.atlassian.net/browse/PP-329#icft=PP-329)
- [PP-428](https://rsklabs.atlassian.net/browse/PP-428)

### CODE REVIEW

File: test/relayHub/RelayHub.test.ts [hunk 1]
```diff
@@ -1,1177 +1,507 @@
...
-//           'no relay workers',
-//           'Relay Server was successfully registered'
-//         );
-//       });
-//     });
-//   }
-// );
+  }
+);
+
```
Comment (AndresQuijano) [inline test/relayHub/RelayHub.test.ts:507]: Please use lower case for the name of this files.

File: test/relayHub/RelayHub.test.ts [hunk 2]
```diff
@@ -1,1177 +1,507 @@
...
-    it('should require maxWorkerCount > 0', async function() {
-      const promiseOfRevert = deployRelayHubContract([undefined, '0']);
+    afterEach(function () {
+      mockRelayHub = undefined as unknown as MockContract<RelayHub>;
+    });
 
-      await expect(promiseOfRevert).to.have.revertedWith(
-        ERR_HUB_BAD_PARAMS
-      );
+    describe('constructor', function () {
```
Comment (AndresQuijano) [inline test/relayHub/RelayHub.test.ts:54]: Here all the fail scenarios are being tested but not the happy path. You are testing some kind of success in every test case but only when is not rejected with ERR_HUB_BAD_PARAMS.

File: test/relayHub/RelayHub.test.ts [hunk 3]
```diff
@@ -1,1177 +1,507 @@
...
     });
-  });
 
-  describe('', function() {
-    beforeEach(async function() {
-      mockRelayHub = await deployRelayHubContract();
-    });
+    describe('', function () {
+      beforeEach(async function () {
+        mockRelayHub = await deployRelayHubContract();
```
Comment (AndresQuijano) [inline test/relayHub/RelayHub.test.ts:129]: Why is this required here when the same instruction is present at line 47 in a more general beforeEach?

File: test/relayHub/RelayHub.test.ts [hunk 4]
```diff
@@ -1,1177 +1,507 @@
...
+              ERR_WORKER_HAS_MANAGER
+            );
+          });
+
+          it('should fail when adding more workers than maxWorkerCount', async function () {
+            const expectedWorkerCountLimit = 13;
+            await mockRelayHub.setVariable('maxWorkerCount', expectedWorkerCountLimit);
+
+            await mockRelayHub.stakeForAddress(relayManagerAddr, 1000, {
+              value: oneRBTC.mul(expectedWorkerCountLimit),
```
Comment (AndresQuijano) [inline test/relayHub/RelayHub.test.ts:253]: This .mul is not required, oneRBTC is enough (same in line 161)

File: test/relayHub/RelayHub.test.ts [hunk 5]
```diff
@@ -1,1177 +1,507 @@
...
+            const promiseOfWorkers = mockRelayHub
+              .connect(relayManager)
+              .addRelayWorkers(workers);
+
+            await expect(promiseOfWorkers).to.revertedWith(ERR_TOO_MANY_WORKERS);
+          });
+        });
+      });
+
+      describe('disableRelayWorkers', function () {
```
Comment (AndresQuijano) [inline test/relayHub/RelayHub.test.ts:268]: There are some cases missing:
-"invalid quantity of workers"
-Check if Relay Manager is staked
-Disable more than one worker
-"Incorrect Manager"
-Disable a subset of the workers.

File: test/relayHub/RelayHub.test.ts (hunk 1 above)
Comment (kenny08gt) [reply test/relayHub/RelayHub.test.ts:507]: Done

File: test/relayHub/RelayHub.test.ts (hunk 3 above)
Comment (kenny08gt) [reply test/relayHub/RelayHub.test.ts:129]: Done

File: test/relayHub/RelayHub.test.ts (hunk 4 above)
Comment (kenny08gt) [reply test/relayHub/RelayHub.test.ts:253]: Done

File: test/relayHub/RelayHub.test.ts (hunk 5 above)
Comment (kenny08gt) [reply test/relayHub/RelayHub.test.ts:268]: Added as to do. Good catch.

File: test/relayHub/RelayHubContractScenarios.test.ts [hunk 6]
```diff
@@ -0,0 +1,737 @@
...
+    relayOwnerAddr = relayOwner.address;
+
+    fakePenalizer = await smock.fake('Penalizer');
+  });
+
+  afterEach(function () {
+    mockRelayHub = undefined as unknown as MockContract<RelayHub>;
+  });
+
+  describe('Manager tests - Stake related scenarios', function() {
```
Comment (AndresQuijano) [inline test/relayHub/RelayHubContractScenarios.test.ts]: This whole describe is duplicated in the RelayHub.test.ts file.

File: test/relayHub/RelayHubContractScenarios.test.ts [hunk 7]
```diff
@@ -0,0 +1,737 @@
...
+    });
+
+    afterEach(function () {
+      mockRelayHub = undefined as unknown as MockContract<RelayHub>;
+    });
+
+
+    it('Should fail a relayRequest if the manager is in the last block of delay blocks', async function() {
+      // let mockRelayHub = await createContractDeployer([fakePenalizer.address]);
+      // const {smartWallet} = await loadFixture(prepareFixture);
```
Comment (AndresQuijano) [inline test/relayHub/RelayHubContractScenarios.test.ts]: Please remove these commented lines.

File: test/relayHub/RelayHubContractScenarios.test.ts [hunk 8]
```diff
@@ -0,0 +1,737 @@
...
+
+      // const calculatedAddr = await factory.getSmartWalletAddress(
+      //   gaslessAccount.address,
+      //   "0x000000000000000000000000000000",
+      //   relayRequestMisbehavingVerifier.request.index
+      // );
+
+      // const smartWalletFactory = await ethers.getContractFactory('SmartWallet');
+      // let calculatedAddr: SmartWallet = await smartWalletFactory.deploy();
+      // await token.mint('1', calculatedAddr);
```
Comment (AndresQuijano) [inline test/relayHub/RelayHubContractScenarios.test.ts]: Please delete these commented lines.
Line 693 as well.

File: test/relayHub/RelayHubContractScenarios.test.ts [hunk 9]
```diff
@@ -0,0 +1,737 @@
...
+        from: externalWallet.address,
+        to: ZERO_ADDRESS,
+        data: '0x',
+        tokenContract: token.address,
+        relayHub: mockRelayHub.address,
+        tokenAmount: '1',
+        tokenGas: '50000',
+        recoverer: ZERO_ADDRESS,
+        index: '0',
+        gas: '90000000000000'
```
Comment (AndresQuijano) [inline test/relayHub/RelayHubContractScenarios.test.ts]: There is a type error here. I don't know the solution but I think it should be addressed. Same at line 684.

File: test/relayHub/RelayHubContractScenarios.test.ts (hunk 6 above)
Comment (kenny08gt) [reply test/relayHub/RelayHubContractScenarios.test.ts]: Removed duplicated

File: test/relayHub/RelayHubContractScenarios.test.ts (hunk 7 above)
Comment (kenny08gt) [reply test/relayHub/RelayHubContractScenarios.test.ts]: Removed unnecessary comments

File: test/relayHub/RelayHubContractScenarios.test.ts (hunk 8 above)
Comment (kenny08gt) [reply test/relayHub/RelayHubContractScenarios.test.ts]: Removed unnecessary comments

File: test/relayHub/RelayHubContractScenarios.test.ts (hunk 9 above)
Comment (kenny08gt) [reply test/relayHub/RelayHubContractScenarios.test.ts]: Fixed with an extension of the type declaration. Suggested by @jurajpiar

**Review by ironFe93 — CHANGES_REQUESTED** [review]
There are conflicts with the target branch @kenny08gt

**Review by AndresQuijano — APPROVED** [review]
